src/Message/Body.php

```diff
@@ -22,13 +30,23 @@
     public function getBody()
     {
         if (!isset($this->body)) {
-            $this->body = Stream::open($this->defaultStream, $this->defaultMode);
+            $this->body = $this->createDefaultBody();
         }
         
         return $this->body;
     }
 
     /**
+     * Set the body
+     *
+     * @param StreamInterface $body
+     */
+    protected function setBody(StreamInterface $body)
+    {
+        $this->body = $body;
+    }
```

Comment:
Simply setting the body is the normal behavior is other traits want it do work differently they can overwrite the method.